src/evaluator/index.js

```diff
@@ -27,7 +27,7 @@ class TypedValue {
     }
 
     if (this.type === 'address') {
-      const isChecksum = /[a-f]/.test(this.value)
+      const isChecksum = /[A-F]/.test(this.value)
```

Comment:
I think this regex should have always tested if there was any uppercase char in the address, and only then require that the checksum is correct.